tests/chez/chez028/ExpressionParser.idr

```diff
@@ -11,45 +11,45 @@ import Data.String.Parser.Expression
 
 table : OperatorTable Nat
 table =
-  [ [Infix (do token "^"; pure (power) ) AssocRight]
-  , [ Infix (do token "*"; pure (*) ) AssocLeft ]
-  , [ Infix (do token "+"; pure (+) ) AssocLeft ]
+  [ [ Infix (token "^" *> pure power ) AssocRight]
+  , [ Infix (token "*" *> pure (*)   ) AssocLeft ]
+  , [ Infix (token "+" *> pure (+)   ) AssocLeft ]
   ]
 
 table' : OperatorTable Integer
 table' =
-  [ [ Infix (do token "*"; pure (*) ) AssocLeft
-    , Infix (do token "/"; pure (div) ) AssocLeft
+  [ [ Infix (token "*" *> pure (*)   ) AssocLeft
+    , Infix (token "/" *> pure (div) ) AssocLeft
     ]
-  , [ Infix (do token "+"; pure (+) ) AssocLeft
-    , Infix (do token "-"; pure (-) ) AssocLeft
+  , [ Infix (token "+" *> pure (+) ) AssocLeft
+    , Infix (token "-" *> pure (-) ) AssocLeft
     ]
   ]
 
 mutual
   term : Parser Nat
-  term = (natural <|> expr) <* spaces
+  term = lexeme (natural <|> expr)
          <?> "simple expression"
 
   expr : Parser Nat
   expr = buildExpressionParser Nat table term
 
 mutual
   term' : Parser Integer
-  term' = (integer <|> expr') <* spaces
-         <?> "simple expression"
+  term' = lexeme (integer <|> expr')
+          <?> "simple expression"
 
   expr' : Parser Integer
   expr' = buildExpressionParser Integer table' term'
 
-showRes : Show a => Either String (a, Int) -> IO ()
-showRes res = case res of
+run : Show a => Parser a -> String -> IO ()
```

Comment:
This `run` seems useful. Should it be moved to the library?

The fact we are happily ignoring `rem` also makes me thing that we are in the business
for and `eof : ParserT m ()` that only succeeds if we have reached the end of the file!